File: NamasteDSA/WarmUp/starPatterns.js
function squarePattern(n) {
  for (let i = 0; i < n; i++) {
    let row = "";
    for (let j = 0; j < n; j++) {
      row = "* " + row;
    }
    console.log(row);
  }
}

function firstTrianglePattern(n) {
  for (let i = 0; i < n; i++) {
    let row = "";
    for (let j = 0; j <= i; j++) {
      row = "* " + row;
    }
    console.log(row);
  }
}

function secondTrianglePattern(n) {
  for (let i = 0; i < n; i++) {
    let row = "";
    for (let j = 0; j <= i; j++) {
      row = row + (j + 1) + " ";
    }
    console.log(row);
  }
}

function thirdTrianglePattern(n) {
  for (let i = 0; i < n; i++) {
    let row = "";
    for (let j = 0; j <= i; j++) {
      row = row + (i + 1) + " ";
    }
    console.log(row);
  }
}

function fourthTrianglePattern(n) {
  for (let i = n; i > 0; i--) {
    let row = "";
    for (let j = 0; j < i; j++) {
      row = row + (j + 1) + " ";
    }
    console.log(row);
  }

  console.log();

  for (let i = 0; i < n; i++) {
    let row = "";
    for (let j = 0; j < n - i; j++) {
      row = row + (j + 1) + " ";
    }
    console.log(row);
  }
}

function fifthTrianglePattern(n) {
  for (let i = 0; i < n; i++) {
    let row = "";
    for (let j = 0; j < n - i; j++) {
      row = row + "* ";
    }
    console.log(row);
  }
}

function sixthTrianglePattern(n) {
  for (let i = 0; i < n; i++) {
    let row = "";
    for (let j = 0; j < n - (i + 1); j++) {
      row = row + "  ";
    }
    for (let j = 0; j <= i; j++) {
      row = row + "* ";
    }
    console.log(row);
  }
}

console.log("Square Pattern");
squarePattern(4);
console.log("First Triangle Pattern");
firstTrianglePattern(4);
console.log("Second Triangle Pattern");
secondTrianglePattern(5);
console.log("Third Triangle Pattern");
thirdTrianglePattern(5);
console.log("Fourth Triangle Pattern");
fourthTrianglePattern(5);
console.log("Fifth Triangle Pattern");
fifthTrianglePattern(5);
console.log("Sixth Triangle Pattern");
sixthTrianglePattern(5);

File: NamasteDSA/WarmUp/StarPatterns.java
public class StarPatterns {
    public static void squarePattern(int rows) {
        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < rows; j++) {
                System.out.print("* ");
            }
            System.out.println();
        }
    }

    public static void firstTrianglePattern(int rows) {
        for (int i = 0; i < rows; i++) {
            for (int j = 0; j <= i; j++) {
                System.out.print("* ");
            }
            System.out.println();
        }
    }

    public static void secondTrianglePattern(int rows) {
        for (int i = 0; i < rows; i++) {
            for (int j = 0; j <= i; j++) {
                System.out.print(j + 1 + " ");
            }
            System.out.println();
        }
    }

    public static void thirdTrianglePattern(int rows) {
        for (int i = 0; i < rows; i++) {
            for (int j = 0; j <= i; j++) {
                System.out.print(i + 1 + " ");
            }
            System.out.println();
        }
    }

    public static void fourthTrianglePattern(int rows) {
        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < rows - i; j++) {
                System.out.print(j + 1 + " ");
            }
            System.out.println();
        }
    }

    public static void fifthTrianglePattern(int rows) {
        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < rows - i; j++) {
                System.out.print("* ");
            }
            System.out.println();
        }
    }

    public static void sixthTrianglePattern(int rows) {
        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < rows - (i + 1); j++) {
                System.out.print("  ");
            }
            for (int j = 0; j <= i; j++) {
                System.out.print("* ");
            }
            System.out.println();
        }
    }

    public static void main(String args[]) {
        squarePattern(4);
        System.out.println();
        firstTrianglePattern(4);
        System.out.println();
        secondTrianglePattern(4);
        System.out.println();
        thirdTrianglePattern(4);
        System.out.println();
        fourthTrianglePattern(4);
        System.out.println();
        fifthTrianglePattern(4);
        System.out.println();
        sixthTrianglePattern(4);
    }
}